iotdb/src/main/java/org/apache/iotdb/db/engine/filenode/FileNodeProcessor.java

```diff
@@ -1481,224 +1428,154 @@ private String queryAndWriteDataForMerge(IntervalFileNode backupIntervalFile)
-        }
         break;
       case INT64:
         seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getLong());
-        count++;
-        startTime = endTime = timeValuePair.getTimestamp();
-        if (!startTimeMap.containsKey(deviceId) || startTimeMap.get(deviceId) > startTime) {
-          startTimeMap.put(deviceId, startTime);
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
-        while (seriesReader.hasNext()) {
-          count++;
-          timeValuePair = seriesReader.next();
-          endTime = timeValuePair.getTimestamp();
-          seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getLong());
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
         break;
       case FLOAT:
         seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getFloat());
-        count++;
-        startTime = endTime = timeValuePair.getTimestamp();
-        if (!startTimeMap.containsKey(deviceId) || startTimeMap.get(deviceId) > startTime) {
-          startTimeMap.put(deviceId, startTime);
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
-        while (seriesReader.hasNext()) {
-          count++;
-          timeValuePair = seriesReader.next();
-          endTime = timeValuePair.getTimestamp();
-          seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getFloat());
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
         break;
       case DOUBLE:
         seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getDouble());
-        count++;
-        startTime = endTime = timeValuePair.getTimestamp();
-        if (!startTimeMap.containsKey(deviceId) || startTimeMap.get(deviceId) > startTime) {
-          startTimeMap.put(deviceId, startTime);
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
-        while (seriesReader.hasNext()) {
-          count++;
-          timeValuePair = seriesReader.next();
-          endTime = timeValuePair.getTimestamp();
-          seriesWriterImpl
-              .write(timeValuePair.getTimestamp(), timeValuePair.getValue().getDouble());
-        }
-        if (!endTimeMap.containsKey(deviceId) || endTimeMap.get(deviceId) < endTime) {
-          endTimeMap.put(deviceId, endTime);
-        }
         break;
       case TEXT:
         seriesWriterImpl.write(timeValuePair.getTimestamp(), timeValuePair.getValue().getBinary());
-        count++;
```

Comment:
这些下面的操作都不要了？

Reply:
1. I think the function name `writeOneSeries()` is confusing. Why the function need a parameter `seriesReader`? (Acutally, the writeOneSeries() is used for the mergence process, so `seriesReader` is useful).
2. I think what MyXOF concerned is why you delete the maintenance of `startTimeMap` and  `endTimeMap`

Reply:
1. To write anything, we need a data source, which is the seriesReader.
2. If you look carefully, you will find that I did not delete the maintenance. To update start time, all we need is the first TVPair and we only need the last TVPair to update end time, which is done in writeOneSeries().

Reply:
> 1. To write anything, we need a data source, which is the seriesReader.

I do not think so. You have a parameter called `timeValuePair`, It is what I think will be write into a series...But after reading the function, I found that I was wrong. That is why I felt `seriesReader` is strange in this function (at least for a function that has this name)

> 2. If you look carefully, you will find that I did not delete the maintenance. To update start time, all we need is the first TVPair and we only need the last TVPair to update end time, which is done in writeOneSeries().
OK.

Reply:
Actually, parameter timeValuePair is the first TVPair of the series. It may be used to determine the start time of the merged file, so it must be read ahead, while it is not yet written, so it is also passed to this function.
I changed it to firstTVPair if that will make it easier to understand.